crates/query-engine/translation/src/translation/query/fields.rs

```diff
@@ -318,3 +323,135 @@ fn translate_nested_field(
         },
     ))
 }
+
+#[allow(clippy::too_many_arguments)]
+/// In order to return the expected type representation for each column,
+/// we need to wrap columns in type representation cast, and unpack composite types
+/// so we can wrap them.
+fn unpack_and_wrap_fields(
+    env: &Env,
+    state: &mut State,
+    current_table: &TableNameAndReference,
+    join_relationship_fields: &mut Vec<relationships::JoinFieldInfo>,
+    column: &str,
+    alias: sql::ast::ColumnAlias,
+    type_info: &CollectionOrCompositeTypeInfo<'_>,
+    nested_field_joins: &mut Vec<JoinNestedFieldInfo>,
+) -> Result<(sql::ast::ColumnAlias, sql::ast::Expression), Error> {
```

Comment:
I actually pulled it out because that function has gotten quite big and found this to be more digestible, but we can talk about this if you really prefer it to be inlined.